ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/StateProcessor.java

```diff
@@ -61,15 +62,20 @@
   private static final ALogger STDOUT = new ALogger("stdout");
   private List<DepositWithIndex> deposits;
   private BeaconStateWithCache initialState;
+  private AttestationAggregator attestationAggregator;
+  private BlockAttestationsPool blockAttestationsPool;
```

Comment:
It occurs to me that both `attestationAggregator` and `blockAttestationsPool` are really only relevant to validators.  Would be nice if we could remove these instances from the `StateProcessor`, and manage the updates to these instances by watching for relevant events in the `ValidatorCoordinator`.

Reply:
But we call methods of both in StateProcessor, in onAttestation?